### PR TITLE
RHDEVDOCS-5518: Documenting Pipelines templates in the admin console

### PR DESCRIPTION
**Pull request by abrennan89**
Version(s):
4.10+

Issue:
https://issues.redhat.com/browse/RHDEVDOCS-5518

Link to docs preview:
- https://62463--docspreview.netlify.app/openshift-enterprise/latest/cicd/pipelines/working-with-pipelines-web-console.html
- https://62463--docspreview.netlify.app/openshift-enterprise/latest/cicd/pipelines/working-with-pipelines-web-console.html#op-creating-pipeline-templates-admin-console_working-with-pipelines-web-console

QE review:
- [x] QE has approved this change.

Additional information:
- Updated name of the assembly and structure so that it refers to "web console" at the top level instead of "Developer perspective", and can include content for both Developer and Administrator perspectives until restructuring work is completed by @mramendi 
- Removed old, outdated content
- CC @Preeticp

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
LGTM!

**Comment by sanketpathak** [comment]
QE Approved!

File: cicd/pipelines/working-with-pipelines-web-console.adoc
Comment (mramendi) [inline cicd/pipelines/working-with-pipelines-web-console.adoc:9]: s/PipelineRuns/PipelineRun/ , because you already have "objects" for the plural

**Comment by abrennan89** [comment]
Waiting for QE review from the pipelines team at request of the pipelines docs team

**Comment by abrennan89** [comment]
Due to removal of content and addition of the new module, this may require a manual cherrypick to any new pipelines branches resulting from pipelines splitting into its own doc set.

@mramendi can you provide the branch name(s) you'd like me to cherrypick this to for the new pipelines standalone docs?

**Comment by mramendi** [comment]
@abrennan89 the branches are pipelines-docs and pipelines-docs-1.11

**Review by mramendi — APPROVED** [review]
I have reviewed the changes and I see nothing wrong from a writing or structure/consistency perspective.

Please do ensure that Pipelines QE, and ideally also Pipelines Development, sign off on the exclusion of the old procedure and on the details of the new one.

In particular, it looks weird to me that pipeline templates get created in the `openshift` namespace when normally pipelines work tends to be in the `openshift-pipelines` namespace. But maybe it is correct. Dev or QE will know for sure.

**Comment by abrennan89** [comment]
@mramendi I would guess that it's in that namespace because it's an admin procedure to create a template, and it needs to be in there so that users can use it in any namespace on the cluster, but hopefully QE can confirm.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (ppitonak) [inline modules/op-creating-pipeline-templates-admin-console.adoc:9]: Original description IMHO explained better why would somebody do it
> As a cluster administrator, to create and deploy an application from a Git repository, you can use custom pipeline templates that override the default pipeline templates provided by {pipelines-title}

File: modules/op-using-custom-pipeline-template-for-git-import.adoc
Comment (ppitonak) [inline modules/op-using-custom-pipeline-template-for-git-import.adoc:34]: IMHO lines 34 to 65 are very valuable, the new version doesn't say anything about removing/disabling the templates that are created by the operator.

Comment (mramendi) [reply modules/op-using-custom-pipeline-template-for-git-import.adoc:34]: @ppitonak could you please explain in what use cases it is useful to remove/disable the templates that are created by the operator? Thanks!

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (abrennan89) [reply modules/op-creating-pipeline-templates-admin-console.adoc:9]: @mramendi +1 that's what I'm wondering, why can't they have both on a cluster?

File: modules/op-using-custom-pipeline-template-for-git-import.adoc
Comment (abrennan89) [reply modules/op-using-custom-pipeline-template-for-git-import.adoc:34]: Per the discussion https://redhat-internal.slack.com/archives/CHG0KRB7G/p1690564699338199?thread_ts=1690217277.713569&cid=CHG0KRB7G - pretty sure that **Tekton Configuration** tab doesn't exist anymore in the Operator (at least it didn't when I tried this on a cluster), or else there's a step missing somewhere that makes it available. This being seemingly outdated was the reason I suggested removing the module.
Also, wouldn't the Operator just recreate these templates if there was an update or something? I might be wrong, I've just seen instances in the past where you didn't modify certain configs because the Operator would just override them anyway.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (ppitonak) [reply modules/op-creating-pipeline-templates-admin-console.adoc:9]: I guess that they can have both but our customers don't want those default templates. @vdemeester can you confirm?

File: modules/op-using-custom-pipeline-template-for-git-import.adoc
Comment (ppitonak) [reply modules/op-using-custom-pipeline-template-for-git-import.adoc:34]: You are right that UI for Tekton Config doesn't exist anymore but customers can still edit it using `oc edit tektonconfig config` or in UI `Administration -> CustomResourceDefinitions -> TektonConfig -> Instances -> config`

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (abrennan89) [reply modules/op-creating-pipeline-templates-admin-console.adoc:9]: @ppitonak @vdemeester can you please confirm if there is anything here worth blocking the PR over, or can it be addressed in a future Jira, etc?

File: modules/op-using-custom-pipeline-template-for-git-import.adoc
Comment (abrennan89) [reply modules/op-using-custom-pipeline-template-for-git-import.adoc:34]: Per @mramendi 's comment
> @ppitonak could you please explain in what use cases it is useful to remove/disable the templates that are created by the operator? Thanks!

It seems like there still isn't an answer for this. This PR seems to be blocked by something that I think could maybe be a separate discussion, i.e. if removing or disabling these isn't crucial in the scheme of things wrt admins creating templates, maybe it should be removed for now and revisited if it is actually requested by users or needed later?
Right now it seems like it's something that's just an option or a preference as opposed to something that's required, and I would like to get this PR closed since it has been open for weeks now.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (vdemeester) [reply modules/op-creating-pipeline-templates-admin-console.adoc:9]: I am actually not sure 😅 I tend to like @ppitonak suggestion better, but I don't have strong opinion 😓 and I don't think we need to block the PR on this 🙃

Comment (ppitonak) [reply modules/op-creating-pipeline-templates-admin-console.adoc:9]: Yup, let's not block this PR

File: modules/op-using-custom-pipeline-template-for-git-import.adoc
Comment (ppitonak) [reply modules/op-using-custom-pipeline-template-for-git-import.adoc:34]: I don't know why would they disable default templates. You are right that this can be solved separately, not a blocker for this PR.

**Review by ppitonak — APPROVED** [review]

File: cicd/pipelines/customizing-configurations-in-the-tektonconfig-cr.adoc
Comment (mramendi) [inline cicd/pipelines/customizing-configurations-in-the-tektonconfig-cr.adoc:55]: Is there a particular reason you did not add a link to the new content here instead, and just removed the old one?

File: cicd/pipelines/working-with-pipelines-web-console.adoc
Comment (mramendi) [inline cicd/pipelines/working-with-pipelines-web-console.adoc:9]: It seems to me that it's either "use the Administrator and Developer perspectives" or  "use the Administrator or Developer perspective"

File: cicd/pipelines/working-with-pipelines-web-console.adoc
Comment (mramendi) [inline cicd/pipelines/working-with-pipelines-web-console.adoc:17]: A level 3 heading right after a level 1 heading?

Comment (mramendi) [reply cicd/pipelines/working-with-pipelines-web-console.adoc:17]: (Even if it is justified, I have a suspicion this might break portal builds, so if you need to keep it we probably want a portal test)

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [inline modules/op-creating-pipeline-templates-admin-console.adoc:9]: Please double-check the use of "which" re IBM SG. In the PDF copy that I have, "that" and "which" are present, and I don't think this particular usage works according to them. I am not 100% sure, however.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [inline modules/op-creating-pipeline-templates-admin-console.adoc:13]: "have switched" is the present perfect tense, while the simple past tense is preferred, or at least that's what I remember to be the case. "Tenses" in the IBM SG just talks of using the present and the simple past.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [inline modules/op-creating-pipeline-templates-admin-console.adoc:14]: Same here re tense

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [inline modules/op-creating-pipeline-templates-admin-console.adoc:40]: After the creation of the pipeline" would work here in my view.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [reply modules/op-creating-pipeline-templates-admin-console.adoc:13]: https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (abrennan89) [reply modules/op-creating-pipeline-templates-admin-console.adoc:40]: I think it's fine as is since this is really about personal preference?

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (abrennan89) [reply modules/op-creating-pipeline-templates-admin-console.adoc:13]: This sentence in my opinion would read very strangely if "have" is removed (presumably that's the suggestion?).

Sometimes there needs to be a little wiggle room on the tenses. For example, in OCP docs generally, we don't allow the use of "will", but in the IBM style guide one of the examples shows how sometimes this can't be avoided, so that the sentence reads properly:
image

I would leave these as they are, especially since this PR has been open for over about 6 weeks now, and I'd prefer not to block merging on anything that isn't a serious issue or is merely a preference in terms of reviews.

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (abrennan89) [reply modules/op-creating-pipeline-templates-admin-console.adoc:14]: See previous comment

File: cicd/pipelines/working-with-pipelines-web-console.adoc
Comment (abrennan89) [reply cicd/pipelines/working-with-pipelines-web-console.adoc:17]: I think it's just a typo, no big deal

File: cicd/pipelines/customizing-configurations-in-the-tektonconfig-cr.adoc
Comment (abrennan89) [reply cicd/pipelines/customizing-configurations-in-the-tektonconfig-cr.adoc:55]: Updated in next commit

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [reply modules/op-creating-pipeline-templates-admin-console.adoc:13]: Agreed on this one

File: modules/op-creating-pipeline-templates-admin-console.adoc
Comment (mramendi) [reply modules/op-creating-pipeline-templates-admin-console.adoc:14]: In plain simple prereqs removing "have" is usually the way to go, but this is minor, I will leave this to you

**Comment by mramendi** [comment]
LGTM. There are further improvements possible but I see how they are hard without another review cycle. I left one question about tenses, but you decide.

**Review by mramendi — APPROVED** [review]

**Comment by abrennan89** [comment]
Pipelines docs cherrypicks followed up from https://github.com/openshift/openshift-docs/pull/64318 instead of CPing directly from this branch to reduce conflicts

**Pull request merged**
